ros2param/test/test_cli.py

```diff
@@ -0,0 +1,365 @@
+                            node_name='_hidden_param_node',
+                            additional_env=additional_env
+                        ),
+                        OpaqueFunction(function=lambda context: ready_fn())
+                    ],
+                    additional_env=additional_env
+                )
+            ]
+        ),
+    ])
+
+
+class TestROS2ParamCLI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(
+        cls,
+        launch_service,
+        proc_info,
+        proc_output,
+        rmw_implementation
+    ):
+        @contextlib.contextmanager
+        def launch_param_command(self, arguments):
+            param_command_action = ExecuteProcess(
+                cmd=['ros2', 'param', *arguments],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2param-cli',
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, param_command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    # ignore ros2cli daemon node
+                    filtered_patterns=['.*ros2cli.*'],
+                    filtered_rmw_implementation=rmw_implementation
+                )
+            ) as param_command:
+                yield param_command
+        cls.launch_param_command = launch_param_command
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_verb(self):
+        parameters = param_node.ParamNode.get_node_parameters()
+        for param_key, param_value in parameters.items():
+            with self.subTest(param_key=param_key, param_value=param_value):
+                with self.launch_param_command(
+                        arguments=['get', '/param_node', param_key]) as param_command:
+                    assert param_command.wait_for_shutdown(timeout=5)
+                assert param_command.exit_code == launch_testing.asserts.EXIT_OK
+                if param_value is None:
+                    param_value = ''
+                assert launch_testing.tools.expect_output(
+                    expected_text=GET_VERB_PARAM_VALUES[param_key].format(
+                        str(param_value)),
+                    text=param_command.output,
+                    strict=True
+                )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_verb_hide_type(self):
+        parameters = param_node.ParamNode.get_node_parameters()
+        for param_key, param_value in parameters.items():
+            with self.subTest(param_key=param_key, param_value=param_value):
+                with self.launch_param_command(
+                        arguments=['get', '/param_node',
+                                   param_key, '--hide-type']) as param_command:
```

Comment:
@BMarchi nit: consider normalizing format. On other places we use something like:

```python
with self.launch_some_command(
    arguments=[
        'get',
        # ...
    ]
) as some_command:
    # ...
```

I don't mind the format as long as we comply with the [style guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#python) and we're consistent about it.

Reply:
This one still applies (throughout all tests I mean).